eng/common/templates/steps/perf-send-to-helix.yml

```diff
@@ -24,7 +24,7 @@ parameters:
 steps:
 - template: /eng/pipelines/common/templates/runtimes/send-to-helix-inner-step.yml
   parameters:
-    osGroup: $(Agent.Os)
+    osGroup: $(Agent.OS)
```

Comment:
YAML identifiers should be case insensitive. I think you need to use variables['Agent.Os'] but use a different substitution syntax. I'll look it up after our team meeting, I think it's something like $[[ (instead of ${{) that expands later so that it "waits" until the variable gets populated. @DrewScoggins might remember the syntax off the top of his head.

Reply:
Is it possible if we directly use this variable as ```variables['Agent.Os']``` in https://github.com/dotnet/runtime/blob/095628568dd6bc3236d1d1eaa1ac97fe9470152a/eng/pipelines/common/templates/runtimes/send-to-helix-inner-step.yml#L10
instead of passing it as a parameter?

Reply:
$[, see

https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops

Reply:
I think it should be $[ variables('Agent.Os') ], at least I hope so

Reply:
Well, the problem is that the inner-step is called from three different scripts, we would need to double-check that variables['Agent.Os'] work everywhere, it can likely break for cross-targeting builds.

Reply:
Thanks for sending the doc! I'll test ```$[ variables('Agent.Os') ]``` first

Reply:
From the log it seems the ```$[ variables['Agent.Os'] ]``` is not expanded before evaluating the condition:
```
Begin evaluating template '/eng/pipelines/common/templates/runtimes/send-to-helix-inner-step.yml'
Evaluating: eq(parameters['osGroup'], 'Windows_NT')
Expanded: eq('$[ variables[''Agent.Os''] ]', 'Windows_NT')
Result: False
Evaluating: ne(parameters['osGroup'], 'Windows_NT')
Expanded: ne('$[ variables[''Agent.Os''] ]', 'Windows_NT')
Result: True
```

Reply:
It seems that the problem is.. the variable won't be populated until runtime but the condition evaluation happens at compile time?  I  was thinking we might want to try using runtime expression here ```$[ ]``` instead of compile expression https://github.com/dotnet/runtime/blob/095628568dd6bc3236d1d1eaa1ac97fe9470152a/eng/pipelines/common/templates/runtimes/send-to-helix-inner-step.yml#L10 @trylek

Reply:
I found another place that uses this condition https://github.com/dotnet/runtime/blob/bb921d6566feb3c06a65a149a01a779072808779/eng/pipelines/common/xplat-setup.yml#L45 
but in this case it works because the variable's ready at compile time 
https://github.com/dotnet/runtime/blob/d992e0cd37d538957ca0d2031e02a24474c091d7/eng/pipelines/common/platform-matrix.yml#L36